blivet/fstab.py

```diff
@@ -0,0 +1,792 @@
+        else:
+            entry.spec = getattr(device, "fstab_spec", None)
+
+        entry.file = None
+        if device.format.mountable:
+            entry.file = device.format.mountpoint
+        if device.format.type == "swap":
+            entry.file = "swap"
+
+        entry.vfstype = device.format.type
+
+        return entry
+
+    def read(self, src_file=""):
+        """ Read the fstab file from path stored in self.src_file. Setting src_file parameter overrides
+            it with new value.
+
+            :keyword src_file: When set, reads fstab from path specified in it
+            :type src_file: str
+        """
+
+        # Reset table
+        self._table = Table()
+        self._table.enable_comments(True)
+
+        # resolve which file to read
+        if src_file == "":
+            if self.src_file is None:
+                # No parameter given, no internal value
+                return
+        elif not src_file:
+            return
+        else:
+            self.src_file = src_file
+
+        self._table.errcb = self._parser_errcb
+        try:
+            self._table.parse_fstab(self.src_file)
+        except Exception as e:  # pylint: disable=broad-except
+            # libmount throws general Exception. Okay...
+            if str(e) == "No such file or directory":
+                log.warning("Fstab file '%s' does not exist", self.src_file)
+            else:
+                raise
+
+    def find_device(self, blivet, spec=None, mntops=None, blkid_tab=None, crypt_tab=None, *, entry=None):
+        """ Find a blivet device, based on spec or entry. Mount options can be used to refine the search.
+            If both entry and spec/mntops are given, spec/mntops are prioritized over entry values.
+
+            :param blivet: Blivet instance with populated devicetree
+            :type blivet: :class: `Blivet`
+            :keyword spec: searched device specs (see man fstab(5) fs_spec)
+            :type spec: str
+            :keyword mntops: list of mount option strings (see man fstab(5) fs_mntops)
+            :type mnops: list
+            :keyword blkid_tab: Blkidtab object
+            :type blkid_tab: :class: `BlkidTab`
+            :keyword crypt_tab: Crypttab object
+            :type crypt_tab: :class: `CryptTab`
+            :keyword entry: fstab entry with its spec (and mntops) filled as an alternative input type
+            :type: :class: `FSTabEntry`
+            :returns: found device or None
+            :rtype: :class: `~.devices.StorageDevice` or None
+        """
+
+        _spec = spec or (entry.spec if entry is not None else None)
+        _mntops = mntops or (entry.mntops if entry is not None else None)
+        _mntops_str = ",".join(_mntops) if mntops is not None else None
+
+        return blivet.devicetree.resolve_device(_spec, options=_mntops_str, blkid_tab=blkid_tab, crypt_tab=crypt_tab)
+
+    def get_device(self, blivet, spec=None, file=None, vfstype=None,
```

Comment:
Why do we have two very similar functions `find_device` and `get_device`? Is there a use case for creating `FileDevice` and `DirectoryDevice` devices that are not in devicetree and returning them?

Reply:
`get_device` is a method taken from anaconda. I am preserving it for compatibility reasons.